throttle.js

```diff
@@ -25,8 +35,8 @@ export default function (delay, noTrailing, callback, debounceMode) {
 	let timeoutID;
 	let cancelled = false;
 
-	// Keep track of the last time `callback` was executed.
-	let lastExec = 0;
+	// `callback` execution will be prevented until `preventUntil`
+	let preventUntil = 0;
```

Comment:
How does this change previous functionality?

Reply:
I changed it as refactoring so I hope it doesn't change any previous functionality.
But I have forgotten to check the actual behavior but the automated tests.
I'll see it.

Reply:
I still don’t understand change 😄 

Can we have `lastExec` for previous functionality and `preventUntil` for new functionality? Or is that duplicating?
I don’t want to make changes to something where we are not sure if it works.
Are automated tests okay for this? Do you think we could do something to additionally check this?

Reply:
I'm sorry if my sentences are not clear.
I'm not an English speaker and my native language is far from English... 😢

Reply:
> Even I myself think this change is complicated much, too 🤯.

So maybe we should see if we can make it simpler :smile:

> The disadvantage is that we will update lastExec when we skip leading execution, although we won't execute callback.

Is this disadvantage only in understanding code or also in functionality?

> Existing automated tests might be insufficient to be sure.
> And it seems to be difficult to write solid test since throttle depends on time.

Existing `throttle` tests don’t help much?

> I'm sorry if my sentences are not clear.

No problem, I understand you perfectly, I’m also not native speaker.

---

The thing is, I would like to avoid introducing a lot of core functionality changes if we don’t fully understand what does changes do. To be fair, I don’t fully understand what’s under the hood here since it’s not my code originally and through time it changed much, but if we’re making changes I would like them to be understandable enough, especially by those who are making them.

Reply:
> So maybe we should see if we can make it simpler 😄

> I would like to avoid introducing a lot of core functionality changes if we don’t fully understand what those changes do. 

OK, I will revive `lastExec` 👍 

> Is this disadvantage only in understanding code or also in functionality?

It is only in understanding code.

> Existing `throttle` tests don’t help much?

They help to prevent some regression.
However, it might be better to have more assertions in them I think, because the code (not only this PR, but also master branch) is highly complicated.
May I add assertions in the tests in another PR before merging this PR to ease worries about unintended changes in previous functionality?

Reply:
I revived `lastExec`.
The current diff is easier than previous one.

Reply:
@seiyab this is much more understandable! I’ve left some comments.